File: src/components/UserInfo.js
import restCalls from "../restCalls"
import { Typography } from "@mui/material";
import react from 'react';
import { useEffect } from "react";

export default function UserInfo() {

    const [username, setUsername] = react.useState("")
    const [email, setEmail] = react.useState("")
    const [address, setAddress] = react.useState("")
    const [landphone, setLandphone] = react.useState("")
    const [mobilephone, setMobilephone] = react.useState("")
    const [nif, setNif] = react.useState("")

    useEffect(() => {
        if(localStorage.getItem('user') === null)
            restCalls.userInfo()
        var user = JSON.parse(localStorage.getItem('user'))
        setUsername(user.username)
        setEmail(user.email)
        setAddress(user.address)
        setLandphone(user.landphone)
        setMobilephone(user.mobilephone)
        setNif(user.nif)


    })

    return (
       <div>
       <p><Typography> Username: {username}</Typography></p> 
       <p><Typography> Email: {email}</Typography></p>
        <Typography> Address: {address}</Typography>
        <Typography> LandPhone: {landphone}</Typography>
        <Typography> MobilePhone: {mobilephone}</Typography>
        <Typography> NIF: {nif}</Typography>
        </div>
       

    )
}

File: AvIndividual/src/main/java/pt/unl/fct/di/adc/avindividual/util/UserInfo.java
package pt.unl.fct.di.adc.avindividual.util;

public class UserInfo {

	public String username;
	public String email;
	public String name;
	public String landphone;
	public String mobilephone;
	public String address;
	public String nif;
	public String role;
	public String state;

	public UserInfo() { }
	
	public UserInfo(String username, String email, String name, String landphone, String mobilephone, String address, String nif, String role, String state) {
		this.username = username;
		this.email = email;
		this.name = name;
		this.landphone = landphone;
		this.mobilephone = mobilephone;
		this.address = address;
		this.nif = nif;
		this.role = role;
		this.state = state;
	}
}